public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -459,46 +468,59 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
     }
 
     return getBackendSrv()
-      .datasourceRequest(req)
-      .then(
-        (result: any) => {
+      .fetch(req)
+      .pipe(
+        map((result: any) => {
           const { data } = result;
           if (data) {
             data.executedQueryString = q;
             if (data.results) {
               const errors = result.data.results.filter((elem: any) => elem.error);
+
               if (errors.length > 0) {
-                throw {
-                  message: 'InfluxDB Error: ' + errors[0].error,
-                  data,
-                };
+                return throwError(
+                  this.handleErrors({
+                    message: 'InfluxDB Error: ' + errors[0].error,
+                    data,
+                  })
+                );
```

Comment:
This is returning an Observable but the other path is returning an array. Maybe use a simple `throw` instead?